### PR TITLE
changed dockerfile naming convention

### PR DESCRIPTION
**Pull request by anoipm**
**Description**

Changes proposed in this pull request:

- changed dockerfile naming convention - now we have the original runtime name (without extra dash inside)

**Related issue(s)**
See also #15016

### CODE REVIEW

File: components/function-controller/internal/controllers/serverless/runtime/runtime_test.go [hunk 1]
```diff
@@ -59,21 +59,21 @@ func TestGetRuntimeConfig(t *testing.T) {
...
+				DockerfileConfigMapName: "dockerfile-nodejs16",
 				RuntimeEnvs: []corev1.EnvVar{
 					{Name: "FUNC_RUNTIME", Value: "nodejs16"}},
 			},
 		},
 		"default": {
-			name:    "nodejs14",
-			runtime: serverlessv1alpha2.NodeJs14,
+			name:    "nodejs12",
+			runtime: "default",
```
Comment (dbadura) [inline components/function-controller/internal/controllers/serverless/runtime/runtime_test.go:69]: The default shouldn't be a case anymore.

**Review by dbadura — CHANGES_REQUESTED** [review]

File: components/function-controller/internal/controllers/serverless/runtime/runtime.go [hunk 2]
```diff
@@ -17,63 +19,45 @@ type Config struct {
...
-			Runtime:                 serverlessv1alpha2.Python39,
-			DependencyFile:          "requirements.txt",
-			FunctionFile:            "handler.py",
-			DockerfileConfigMapName: "dockerfile-python-39",
-			RuntimeEnvs: []corev1.EnvVar{
+		config.DependencyFile = "requirements.txt"
+		config.FunctionFile = "handler.py"
+		break
+	}
+	switch runtime {
```
[the remark this answers is not in the document]
Comment (anoipm) [reply components/function-controller/internal/controllers/serverless/runtime/runtime.go:36]: done

File: components/function-controller/internal/controllers/serverless/runtime/runtime_test.go (hunk 1 above)
Comment (anoipm) [reply components/function-controller/internal/controllers/serverless/runtime/runtime_test.go]: Done after discussion with Filip

File: components/function-controller/README.md [hunk 3]
```diff
@@ -22,31 +22,31 @@ To develop the component, use the formulae declared in the [generic](/common/mak
...
+| **APP_FUNCTION_BUILD_LIMITS_MEMORY**                      | Maximum amount of memory assigned to the Job to build a Function image. See [this](https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/#meaning-of-cpu) document for available values.                                                                                      | `1Gi`                                                                                                                                                   |
+| **APP_FUNCTION_BUILD_RUNTIME_CONFIG_MAP_NAME**            | Name of the ConfigMap that contains the runtime definition                                                                                                                                                                                                                                                   | `dockerfile-nodejs12`                                                                                                                                   |
```
[the remark this answers is not in the document]
Comment (anoipm) [reply components/function-controller/README.md:42]: I don't know anything about these envs. I only changed `nodejs-12` to `nodejs12` :(

**Review by pPrecel — APPROVED** [review]
LGTM!

/meow

File: components/function-controller/README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (anoipm) [reply components/function-controller/README.md:42]: I created issue for this.
See also: #15341

**Review by grego952 — APPROVED** [review]

**Pull request merged**